content/_data/upgrades/2-249-1.adoc

```diff
@@ -120,11 +120,39 @@ io.jenkins.plugins.casc.ConfiguratorException: 'statusFilter' is deprecated
 
 ==== Alternate URL removed from inbound agent
 
-Jenkins 2.249.1 no longer provides a fallback URL in the inbound agent launch file.
+Jenkins 2.249.1 no longer provides a fallback URL in the inbound agent launch file (the "slave-agent.jnlp" file).
 Prior releases allowed the administrator to configure a second URL for inbound agents that was different from the Jenkins root URL.
 If the root URL was unreachable by the agent, the second URL was used.
 
-Jenkins includes automatic reverse proxy configuration warnings if the root URL is not configured or is configured incorrectly.
-An alternate URL to connect to the Jenkins server should only be needed in very rare cases.
+If the Jenkins root URL is not accessible to an inbound agent, the agent must connect through TCP using options available from `agent.jar`.
+TCP inbound agent connection is more complicated than the typical inbound agent connection because it requires the instance identity in addition to the usual inbound agent data.
```

Comment:
Huh? Why are we recommending `-instanceIdentity`? That is super obscure and probably never needed.

* If the agent can see the Jenkins HTTP port under the alternate URL, just connect to that instead. Using `-url` the lower-level launcher form, not `-jnlpUrl`.
* If it cannot access Jenkins HTTP, you can use the `-direct` mode.

Reply:
I misinterpreted your earlier comments to mean that you wanted to avoid HTTP.  I was able to make it work with instance identity and found the instance identity documentation to link for more details.

I'll rework to recommend `-url` as first choice with `-direct` as second choice (after I've tested and confirmed that I can make it work myself).  I'd love to not mention instance identify in this section.

Reply:
Would be great to get it landed, rather sooner than later

Reply:
Reworked in 75c6014ff05854667279f9b61fe56c4d561dae90 .  Much simpler now.

The `-direct` command line argument requires the `-instanceIdentity` command line argument, but since it is unchanged from previous releases, I used a block comment to hide the instructions for `-direct`.  That makes the presentation to the reader much simpler, clearer, and specific to the changes in 2.249.1.

Reply:
@jglick this is ready for another review.